packages/orm/src/EntityManager.ts

```diff
@@ -1225,14 +1225,21 @@ export class EntityManager<C = unknown> {
         todo.push(
           ...relations
             .filter((r) => "get" in r)
-            // We skip recursing into CustomCollections and CustomReferences for two reasons:
+            // We skip recursing into CustomCollections and CustomReferences and PersistedAsyncReferencesImpl for two reasons:
             // 1. It can be tricky to ensure `{ forceReload: true }` is passed all the way through their custom load
             // implementations, and so it's easy to have `.get` accidentally come across a not-yet-loaded collection, and
             // 2. Any custom load functions should use the underlying o2m/m2o/etc relations anyway, so if we crawl/refresh
             // those, then when the user calls `.get` on custom collections/references, they should be talking to always-loaded
             // relations, w/o us having to tackle the tricky bookkeeping problem passing `forceReload` all through their
             // custom load function + any other collections they call.
-            .filter((r) => !(r instanceof CustomCollection || r instanceof CustomReference))
+            .filter(
+              (r) =>
+                !(
+                  r instanceof CustomCollection ||
+                  r instanceof CustomReference ||
+                  r instanceof PersistedAsyncReferenceImpl
+                ),
+            )
```

Comment:
The comment above also applies to `PersistedAsyncReference` when `em.refresh` is called it doesn't propagate properly and ends up throwing a `... is not loaded`

Reply:
@0ces I think this makes sense...can you add a unit test in `EntityManager.test` (it has some tests for `refresh`) to make sure this doesn't regress?